### ISSUE TITLE
type DiffEqArray has no field retcode

### ISSUE THREAD

**Issue by timdkim** [FluxML/Zygote.jl#783]
The following code throws `type DiffEqArray has no field retcode`. The error goes away when `sensealg=ReverseDiffAdjoint()` is changed to default (changing it to `TrackerAdjoint()` gives `not implemented` error).

```
using DifferentialEquations, DiffEqSensitivity, DiffEqFlux, Flux, LinearAlgebra, Optim, Plots

tspan = (0.f0, 10.f0)
teval = range(tspan[1],tspan[2],length=25)
dudt1 = Chain(Dense(2, 2))
p1, re1 = Flux.destructure(dudt1)
θ = p1
u0 = Float32[0.;2.]

function dudt1_(du, u, p, t)
    du .= re1(p[1:length(p1)])(u)
end

prob1 = ODEProblem(dudt1_,u0,tspan)

function predict_n_ode()
    _prob=remake(prob1, p=θ)
    solve(_prob, Tsit5(), u0=u0, p=θ, saveat=teval, sensealg=ReverseDiffAdjoint())
end

function loss_n_ode()
    pred = predict_n_ode()
    loss = 0
    if pred.retcode != :Success && pred.retcode != :Converged
        loss = Inf
    else
        pred = Array(pred)
        loss = sum(pred) # arbitrary loss
    end
    loss
end

ps = Flux.params(θ)
opt = ADAM(0.01)

loss, back = Flux.pullback(ps) do
    loss_n_ode()
end

grad = back(1f0)
Flux.Optimise.update!(opt, ps, grad)
```

**Comment by CarloLucibello** [comment]
The `DiffEqArray has no field retcode` error is gone (retcode is now an enum); the gradient computes cleanly. Closing as fixed.

_(Triaged against Zygote v0.7.10 on Julia 1.12.6.)_
